src/corelibs/U2Algorithm/src/temperature/BaseTempCalc.h

```diff
@@ -0,0 +1,97 @@
+/**
+ * UGENE - Integrated Bioinformatics Tools.
+ * Copyright (C) 2008-2022 UniPro <ugene@unipro.ru>
+ * http://ugene.net
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+#pragma once
+
+#include <QString>
+#include <QMap>
+#include <QVariant>
+
+#include <U2Core/global.h>
+
+namespace U2 {
+
+/**
+ * Settings for the temperature calculations
+ */
+struct U2ALGORITHM_EXPORT TempCalcSettings {
```

Comment:
You expect this struct to be inherited and use it a lot as a pointer to the parent class: 
-> make it a class and add a virtual destructor, so deallocation will work correctly.

Reply:
> add a virtual destructor

I can assume, that I forgot some C++ bases, but why do we need virtual destructor here? We do not have pointers in a base class, so even if some heir won't call it when delete it nothing bad happens.

> make it a class

`class` and `struct` is absolutely the same but the fact that `class` is `private` by default, and `struct` is public. And it's better to use `struct` in case of objects like this - when we have a lot of public members, which could vary independently. [Bjarne Stroustrup said this](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c2-use-class-if-the-class-has-an-invariant-use-struct-if-the-data-members-can-vary-independently).

Reply:
`TempCalcSettings` and heirs are removed, settings are stored in QVariantmap

Reply:
This is good: QVariantMap is the best option here.

About virtual destructors: If B extends A and you do not have a virtual destructor in the code like this: `A* a = createB(); delete a;` the destructor from B won't be called at all (behavior is undefined) -> memory leak.

Reply:
Ok. Are we finished here?

Reply:
Yes. Current variant of the solution is safe and simple